back/boxtribute_server/business_logic/warehouse/box/crud.py

```diff
@@ -131,6 +182,47 @@ def update_box(
     Insert timestamp for modification and return the box.
     """
     box = Box.get(Box.label_identifier == label_identifier)
+    box_contains_measure_product = box.size_id is None
+
+    if product_id is None:  # no product change
```

Comment:
These if statements created bit of a knot in my head and are not super easy to read.
A few comments would be nice for the different cases.